src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size;
+    private int minCapacity;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+        this.minCapacity = DEFAULT_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        this.array = grow(minCapacity);
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndexForAddMethod(index);
+        if (index + 1 <= minCapacity) {
+            this.array = grow(minCapacity);
+            T[] newArray = grow(minCapacity);
+            System.arraycopy(this.array, 0, newArray, 0, index);
+            newArray[index] = value;
+            System.arraycopy(this.array, index, newArray, index + 1, minCapacity - index - 1);
+            copyArray(newArray, this.array);
+            size++;
```

Comment:
I can see 2 array grows and 3 System.arraycopy in this method... we can use only 1 System.arraycopy method here and we don't need grow method here at all.

Reply:
Simplified almost all. I left one grow method call here and one System.arraycopy. If I'll remove this one, I can't grow capacity for this method